### DISCUSSION TITLE
Ignore third party library custom-elements from SSR ?

### DISCUSSION THREAD

**Discussion by KevinCarnaille2** [nuxt/nuxt#18631] [Questions]
Hello,

I'm trying to figure out how to apply the same behavior than `<ClientOnly>` wrapper from a "higher level", basically from the configuration.

I have a third party library which declares custom elements (stenciljs), and using `<ClientOnly>`  is not really the best option for us, even if it's very handy.  
The library is loaded from a `CDN`, called in a nuxt module.

Here is a stackblitz that reproduces the issue (in a simplified context) : [https://stackblitz.com/edit/nuxt-starter-6crc9x?file=app.vue](https://stackblitz.com/edit/nuxt-starter-6crc9x?file=app.vue)

I was thinking of a configuration like so, playing with the vue3 compiler :

```
    vue: {
        compilerOptions: {
          isCustomElement: (tag) => tag.startsWith('my-prefix'),
        },
    }, 
```

But it has no effect, nuxt is still trying to hydrate my custom element, and it leads to warnings like so : 
`Hydration children mismatch in <my-element>: server rendered element contains more child nodes than client vdom.
`

Maybe we can do something with Nitro hooks ?
I'm not sure this is an actual bug, that's why I prefer asking because maybe I'm missing something in the docs ! 

Thank you

**Comment by Baroshem** [comment]
Hey,

Not sure if I understand the case correctly but maybe you could use a client only plugin for that?

Something like this -> https://nuxt.com/docs/guide/directory-structure/plugins#vue-plugins

**Reply by KevinCarnaille2** [reply]
Hello, I've added a stackblitz link in my original message, to help understanding the issue :) 
https://stackblitz.com/edit/nuxt-starter-6crc9x?file=app.vue

**Reply by Baroshem** [reply]
As this is about using components, naturally I would recommend a solution that you have also mentioned in your issue which is ClientOnly. 

Don't have any other ideas right now.
